crates/circuit/src/dag_circuit.rs

```diff
@@ -0,0 +1,6474 @@
+                        e.clone_ref(py)
+                    }
+                })?
+                .into_iter(),
+        )
+    }
+
+    fn is_wire_idle(&self, wire: &Wire) -> PyResult<bool> {
+        let (input_node, output_node) = match wire {
+            Wire::Qubit(qubit) => (self.qubit_input_map[qubit], self.qubit_output_map[qubit]),
+            Wire::Clbit(clbit) => (self.clbit_input_map[clbit], self.clbit_output_map[clbit]),
+            Wire::Var(var) => (
+                self.var_input_map.get(var).unwrap(),
+                self.var_output_map.get(var).unwrap(),
+            ),
+        };
+
+        let child = self
+            .dag
+            .neighbors_directed(input_node, Outgoing)
+            .next()
+            .ok_or_else(|| {
+                DAGCircuitError::new_err(format!(
+                    "Invalid dagcircuit input node {:?} has no output",
+                    input_node
+                ))
+            })?;
+
+        Ok(child == output_node)
+    }
+
+    fn may_have_additional_wires(&self, py: Python, instr: &PackedInstruction) -> bool {
+        let has_condition = match instr.condition() {
+            None => false,
+            Some(condition) => !condition.bind(py).is_none(),
+        };
+
+        if has_condition {
+            return true;
+        }
+        let OperationRef::Instruction(inst) = instr.op.view() else {
+            return false;
+        };
+        inst.instruction
+            .bind(py)
+            .is_instance(imports::CONTROL_FLOW_OP.get_bound(py))
+            .unwrap()
+            || inst
+                .instruction
+                .bind(py)
+                .is_instance(imports::STORE_OP.get_bound(py))
+                .unwrap()
+    }
+
+    fn additional_wires(
+        &self,
+        py: Python,
+        op: OperationRef,
+        condition: Option<&PyObject>,
+    ) -> PyResult<(Vec<Clbit>, Vec<PyObject>)> {
+        let wires_from_expr = |node: &Bound<PyAny>| -> PyResult<(Vec<Clbit>, Vec<PyObject>)> {
+            let mut clbits = Vec::new();
+            let mut vars = Vec::new();
+            for var in imports::ITER_VARS.get_bound(py).call1((node,))?.iter()? {
+                let var = var?;
+                let var_var = var.getattr("var")?;
+                if var_var.is_instance(imports::CLBIT.get_bound(py))? {
+                    clbits.push(self.clbits.find(&var_var).unwrap());
+                } else if var_var.is_instance(imports::CLASSICAL_REGISTER.get_bound(py))? {
+                    for bit in var_var.iter().unwrap() {
+                        clbits.push(self.clbits.find(&bit?).unwrap());
+                    }
+                } else {
+                    vars.push(var.unbind());
+                }
+            }
+            Ok((clbits, vars))
+        };
+
+        // let mut bits = Vec::new();
+        let mut clbits = Vec::new();
+        let mut vars = Vec::new();
```

Comment:
Is the comment left over?

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/12550/commits/2abe0842e0096e7c763f4d9fa25c0c12fa7938ed